File: 4_Sorting_Algorithms/Selection_Sort/Python/selection_sort.py
from typing import List


def selection_sort(arr: List[int]) -> List[int]:
    """
    Sorts an array using the selection sort algorithm.

    :param arr: List of integers to be sorted.
    :return: The sorted list.
    """
    arr_len = len(arr)
    
    for i in range(arr_len):
        smallest_idx = i
        for j in range(i + 1, arr_len):
            if arr[j] < arr[smallest_idx]:
                smallest_idx = j
        
        if i != smallest_idx:
            arr[i], arr[smallest_idx] = arr[smallest_idx], arr[i]
    
    return arr

File: 4_Sorting_Algorithms/Selection_Sort/C/selection_sort.c
#include "selection_sort.h"

/**
 * selection_sort - Sorts an array using the selection sort algorithm.
 *
 * @arr: Pointer to the array to be sorted.
 * @size: The number of elements in the array.
 */
void selection_sort(int arr[], int size){
    int temp;

    for (int i = 0; (i < size - 1); i++){
        int smallest_idx = i;
        for (int j = (i + 1); (j < size); j++){
            if (arr[j] < arr[smallest_idx]){
                smallest_idx = j;
            }
        }
        if (i != smallest_idx){
            temp = arr[i];
            arr[i] = arr[smallest_idx];
            arr[smallest_idx] = temp;
        }
    }
}